h/accounts/views.py

```diff
@@ -38,6 +38,70 @@ def ajax_payload(request, data):
     return payload
 
 
+def to_xhr_response(request, non_xhr_result, form):
+    """
+    Return an XHR response for the given ``form``, or ``non_xhr_result``.
+
+    If the given ``request`` is an XMLHttpRequest then return an XHR form
+    submission response for the given form (contains only the ``<form>``
+    element as an HTML snippet, not the entire HTML page).
+
+    If ``request`` is not an XHR request then return ``non_xhr_result``, which
+    should be the result that the view callable would normally return if this
+    were not an XHR request.
+
+    :param request: the Pyramid request
+
+    :param non_xhr_result: the view callable result that should be returned if
+        ``request`` is *not* an XHR request
+
+    :param form: the form that was submitted
+    :type form: deform.form.Form
+
+    """
+    if not request.is_xhr:
+        return non_xhr_result
+
+    request.response.body = form.render().encode()
```

Comment:
Rather than manually encoding the form content, I think it would probably be better to do something like:

```python
request.override_renderer = 'text'
return form.render()
```

That way details of text encoding, content-type, etc., are left up to Pyramid.